routes/soldiers.js

```diff
@@ -0,0 +1,31 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, soldierData, done) {
+  const parsedUrlString = req.url.split("/")
+  if (parsedUrlString.length > 3) {
+    return done(new Error())
+  }
+  const soldierId = parsedUrlString[2];
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
```

Comment:
If you would parse the url with the urlib before everything - you could use `parsedUrl` instead of `parsedUrlString` - and be more readable and happier!